cloudant-client/src/test/java/com/cloudant/tests/ViewsMockTest.java

```diff
@@ -0,0 +1,149 @@
+package com.cloudant.tests;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.cloudant.client.api.CloudantClient;
+import com.cloudant.client.api.Database;
+import com.cloudant.client.api.views.Key;
+import com.cloudant.client.api.views.SettableViewParameters;
+import com.cloudant.client.api.views.UnpaginatedRequestBuilder;
+import com.cloudant.tests.base.TestWithMockedServer;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.MockResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class ViewsMockTest extends TestWithMockedServer {
+
+
+    /**
+     * We can't test the server behaviour of stale, but we can test the URL values are what we
+     * expect. This test uses the various values and checks the stale parameter in the URL, it makes
+     * a request using getSingleValue() as it is easier to mock the responses.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void staleParameterValues() throws Exception {
+        CloudantClient client = CloudantClientHelper.newMockWebServerClientBuilder(server)
+            .build();
+        Database database = client.database("notarealdb", false);
+        for (String s : new String[]{
+                SettableViewParameters.STALE_NO,
+                SettableViewParameters.STALE_OK,
+                SettableViewParameters.STALE_UPDATE_AFTER}) {
+            UnpaginatedRequestBuilder<String, Integer> viewBuilder = database.getViewRequestBuilder
+                    ("testDDoc", "testView").newRequest(Key.Type.STRING, Integer.class);
+            MockResponse mockResponse = new MockResponse().setResponseCode(200).setBody
+                    ("{\"rows\":[{\"key\":null,\"value\":10}]}");
+            server.enqueue(mockResponse);
+            viewBuilder.stale(s).build().getSingleValue();
+            HttpUrl url = server.takeRequest(1, TimeUnit.SECONDS).getRequestUrl();
+            Assert.assertEquals(s, url.queryParameter("stale"));
+            Assert.assertEquals("/notarealdb/_design/testDDoc/_view/testView", url.encodedPath());
+        }
+    }
+
+    /**
+     * Similar to staleParameterValues, but for `stable` parameter
```

Comment:
I'm wondering if `staleParameterValues` and `staleParameterValues` should have a comment like `Test checks the [stable|stale] parameter in the URL by making a mock request using getSingleValue() ...`